### PR TITLE
Task/comb packages

### LINKED ISSUE #9: Improve fullfil & beautify documentation 🌞
Rewrite documentation to clarify use of packages better. 

* better describe modificators, their usage and examples (we car refer to playground at [Resizin page](resizin.com))
* add better examples of api methods usage

### PR DESCRIPTION
**Pull request by horaklukas**
Fixes #9 

* Improve documentation of image modifiers and `node-resizin` package
* Fix typo in filter possible value, it's `greyscale`, not`grayscale`
* Add documentation builder which wire documentation pieces together into package README
* Make buildUrlFactory a default export from resizin package
* Add contribution guide
* Add husky scripts

### CODE REVIEW

File: docs/Upload.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+### `uploadFactory(options: Options): function(file, imageId?)`
+
+```typescript
+interface Options {
+    serverUrl?: string;
```
Comment (jancama2) [inline docs/Upload.md]: dopsat default server url?

File: packages/node-resizin/README.md [hunk 2]
```diff
@@ -1,35 +1,73 @@
...
+const upload = uploadFactory({
+    serverUrl: 'https://api.resizin.com', 
+    apiKey: config.RESIZIN_API_KEY,
+});
+
+const file = fs.createReadStream(__dirname + '/myfile.png');
 
- - `baseUrl` contains default value which serves resizin production server
- - `type`  can be `image` or ```file``` and defines whether you will work only with images or with files. When `file` is selected you can still upload images. Restrictions are only for `image` type.
+resizin.upload(file).then(() => {
```
Comment (jancama2) [inline packages/node-resizin/README.md]: `resizion` proměnná neexistuje

File: docs/Upload.md (hunk 1 above)
Comment (jancama2) [reply docs/Upload.md]: English: add default server ulr?

File: scripts/compose-docs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env node
```
[the remark this answers is not in the document]
Comment (jancama2) [reply scripts/compose-docs]: English: Did you write it your own or took it somewhere?

[the remark this answers is not in the document]
Comment (horaklukas) [reply scripts/compose-docs]: I told you I'm a Node.js half-proof 😄 I tried to find a tool like that but I failed so I did write it by myself.

File: docs/Upload.md (hunk 1 above)
Comment (horaklukas) [reply docs/Upload.md]: It's mentioned in Usage tips section at the end, but you're right, I mentioned it also here.

**Pull request merged**